rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -209,6 +251,477 @@ impl ParquetExec {
+    // }
+
+    let first_group_stats = statistics.iter().find(|s| s.is_some());
+    let first_group_stats = if let Some(Some(statistics)) = first_group_stats {
+        // found first row group with statistics defined
+        statistics
+    } else {
+        // no row group has statistics defined
+        return build_null_array(data_type, statistics_count);
+    };
+
+    let (data_size, arrow_type) = match first_group_stats {
+        ParquetStatistics::Int32(_) => (std::mem::size_of::<i32>(), DataType::Int32),
+        ParquetStatistics::Int64(_) => (std::mem::size_of::<i64>(), DataType::Int64),
+        ParquetStatistics::Float(_) => (std::mem::size_of::<f32>(), DataType::Float32),
+        ParquetStatistics::Double(_) => (std::mem::size_of::<f64>(), DataType::Float64),
+        ParquetStatistics::ByteArray(_) if data_type == &DataType::Utf8 => {
+            (0, DataType::Utf8)
+        }
+        _ => {
+            // type of statistics not supported
+            return build_null_array(data_type, statistics_count);
+        }
+    };
+
+    let statistics = statistics.iter().map(|s| {
+        s.filter(|s| s.has_min_max_set())
+            .map(|s| match statistics_type {
+                StatisticsType::Min => s.min_bytes(),
+                StatisticsType::Max => s.max_bytes(),
+            })
+    });
+
+    if arrow_type == DataType::Utf8 {
+        let data_size = statistics
+            .clone()
+            .map(|x| x.map(|b| b.len()).unwrap_or(0))
+            .sum();
+        let mut builder =
+            arrow::array::StringBuilder::with_capacity(statistics_count, data_size);
+        let string_statistics =
+            statistics.map(|x| x.and_then(|bytes| std::str::from_utf8(bytes).ok()));
+        for maybe_string in string_statistics {
+            match maybe_string {
+                Some(string_value) => builder.append_value(string_value).unwrap(),
+                None => builder.append_null().unwrap(),
+            };
+        }
+        return Arc::new(builder.finish());
+    }
+
+    let mut data_buffer = MutableBuffer::new(statistics_count * data_size);
+    let mut bitmap_builder = BooleanBufferBuilder::new(statistics_count);
+    let mut null_count = 0;
+    for s in statistics {
+        if let Some(stat_data) = s {
+            bitmap_builder.append(true);
+            data_buffer.extend_from_slice(stat_data);
+        } else {
+            bitmap_builder.append(false);
+            data_buffer.resize(data_buffer.len() + data_size);
+            null_count += 1;
+        }
+    }
+
+    let mut builder = ArrayData::builder(arrow_type)
+        .len(statistics_count)
+        .add_buffer(data_buffer.into());
+    if null_count > 0 {
+        builder = builder.null_bit_buffer(bitmap_builder.finish());
+    }
+    let array_data = builder.build();
+    let statistics_array = make_array(array_data);
+    if statistics_array.data_type() == data_type {
+        return statistics_array;
+    }
```

Comment:
Thank you for your feedback. I was looking for a (mostly) generic approach to building statistics arrays and this is the simplest implementation I could come up with. Your questions make me think if this could be done with generics though.
From what I have seen, parquet statistics are only stored as Int32, Int64, Float, Double or ByteArray (used for strings and other complex types); may be someone with more experience with parquet can advise on how statistics work for boolean columns.

Reply:
@jorgecarleitao actually what is wrote in my previous comment is incorrect - boolean is valid statistics type; although in most cases I suspect that it may not provide very helpful statistics (because it only has two values - true and false); anyway I will look into a better implementation for the `build_statistics_array` method and support for more types, but probably in a separate PR as this one is already fairly large